lib/application.ts

```diff
@@ -1383,4 +1389,21 @@ export class SNApplication {
       return base;
     }
   }
+
+  public async handleExpiredAccessToken() {
```

Comment:
Why is the `application` handling expired tokens? This sounds like a service's internal implementation detail. Application functions, and specifically `public` ones, are meant for client consumption (web/mobile).